model_analyzer/config/generate/automatic_model_config_generator.py

```diff
@@ -68,11 +68,13 @@ def _done_walking(self):
            and self._done_walking_instance_count()
 
     def _step(self):
+        self._step_max_batch_size()
+
         if self._done_walking_max_batch_size():
-            self._reset_max_batch_size()
             self._step_instance_count()
-        else:
-            self._step_max_batch_size()
+
+            if not self._done_walking_instance_count():
```

Comment:
ok I see what is going on. We are incrementing both max_batch_size and instance count (in nested loops). If we reset max_batch_size on the last iteration, then we are no longer "done" according to our is_done() function.

Any reason we can't change is_done to just be based on done_walking_instance_count() instead of on both?

Reply:
I agree with your suggestion. Fixed the part.